### PR TITLE
Add PR checks

### PR DESCRIPTION
**Pull request by charlieegan3**
This adds PR checks for:

- js/json/yaml formatting
- markdown formatting
- cspell checking

### CODE REVIEW

File: .markdownlint.yaml [hunk 1]
```diff
@@ -0,0 +1,275 @@
+# Refer to https://github.com/DavidAnson/markdownlint for additional rules
+
+# Set default state for rules to false in order to handle markdownlint new rule additions
+default: false
+
+# MD001/header-increment - Heading levels should only increment by one level at a time
+MD001: true
+
+# MD003/header-style - Heading style
```
Comment (anderseknert) [inline .markdownlint.yaml:9]: Heh, comments like these (which are absolutely necessary btw) are why I'm so happy I opted for human readable linter rule names in Regal.

File: README.md [hunk 2]
```diff
@@ -1,39 +1,42 @@
-# vscode-opa
+# vscode-OPA
```
Comment (anderseknert) [inline README.md:1]: Not sure about this. I'd either leave it be (as it reflects the name of the repo) or use "VS Code OPA extension" or something like that?

File: README.md [hunk 3]
```diff
@@ -1,39 +1,42 @@
...
+- Diagnostics (linting)
+- Hover / tooltips (for inline docs on built-in functions)
+- Go to definition (ctrl/cmd + click on a reference to go to definition)
+- Folding ranges (expand/collapse blocks, imports, comments)
+- Document and workspace symbols (navigate to rules, functions, packages)
+- Inlay hints (show names of built-in function arguments next to their values)
+- Formatting (`opa fmt`, `opa fmt --rego-v1` or `regal fix`, see [Configuration](#configuration) below)
+- Code actions (quick fixes for linting issues)
+- Code completions
+- Code lenses (click to evaluate any package or rule in the editor, and have the result displayed directly on the same line)
```
Comment (anderseknert) [inline README.md:29]: `-` over `*`? thanks ;)

**Review by anderseknert — APPROVED** [review]
Thanks! Not a huge fan of 2 spaces for indentation of code, but I guess it's some kind of consensus for JS/TS these days..

File: README.md (hunk 2 above)
Comment (charlieegan3) [reply README.md]: Oops, that's a mistake. Good spot!

**Review by anderseknert — APPROVED** [review]

**Pull request merged**